Components with react

tag6071 opened the pull request:
Kindly help review this PR to progress on my checkpoint @osule and @andela-oaladeusi thanks

tag6071 committed:
Still debugging deployment to heroku

tag6071 committed:
API testcases for user signing up and signing in.

tag6071 committed:
Merge pull request #33 from ekundayo-ab/continue-routes-test

Continue routes test

tag6071 committed:
Testing user routes to ensure expected results are gotten.

tag6071 committed:
Merge pull request #34 from ekundayo-ab/continue-routes-test

Testing user routes to ensure expected results are gotten.

tag6071 committed:
update test to reflect removal of unneeded modules.

tag6071 committed:
Testing of API routes still in progress.

tag6071 committed:
Debug travis build.

tag6071 committed:
travis debugging still on.

tag6071 committed:
still debug travis integration

tag6071 committed:
test application against more cases.

tag6071 committed:
Refactor readme to be more descriptive on installation, usage and testing of application.

tag6071 committed:
Merge pull request #36 from ekundayo-ab/continue-routes-test

Continue routes test

tag6071 committed:
put feedback into place by validating against unexpected inputs to the application.

tag6071 committed:
Debugging travis build

tag6071 committed:
Debugging travis build

tag6071 committed:
refactor tests to conform with feedback changes.

tag6071 committed:
debugging travis

tag6071 committed:
Hound Debugging

tag6071 committed:
Debugging travis

tag6071 committed:
Refactor to pass by hound

tag6071 committed:
Merge pull request #37 from ekundayo-ab/feedback-changes

Feedback changes

tag6071 committed:
validate email address on user registration.

tag6071 committed:
Arrange logics to follow proper construct.

tag6071 committed:
Merge pull request #38 from ekundayo-ab/feedback-changes

validate email address on user registration.

tag6071 committed:
Test other books CRUD api routes.

tag6071 committed:
Merge pull request #39 from ekundayo-ab/feedback-changes

Test other books CRUD api routes.

tag6071 committed:
More feedback changes to validate against unexpected inputs.

tag6071 committed:
Merge pull request #40 from ekundayo-ab/feedback-changes

More feedback changes to validate against unexpected inputs.

tag6071 committed:
Extra input field allows to enter ISBN of boook

tag6071 committed:
Refactor for more interactivity and user flow

tag6071 committed:
Refactor links to flow into one another

tag6071 committed:
Refactoring for better views.

tag6071 committed:
Merge pull request #41 from ekundayo-ab/fine-tune-template

Fine tune template

tag6071 committed:
Documenting code

tag6071 committed:
Modify badges in readme.md

tag6071 committed:
Merge pull request #42 from ekundayo-ab/code-documentation

Code documentation

tag6071 committed:
Abstract validation logic for admin and inputs, fix wrong logic in showing borrowed but not returned.

tag6071 committed:
Helper function to check if user is admin and also to validate inputs.

tag6071 committed:
travis temporary database configuration.

tag6071 committed:
Merge branch 'code-documentation' into develop

tag6071 committed:
merging and fixing conflicts

tag6071 committed:
travis ci fix

tag6071 committed:
fix travis database configuration

tag6071 committed:
strange abnormalities with sequelize accepting null values, even after defining not null.

tag6071 committed:
Validate user inputs and ensure code base is documented for better description.

tag6071 committed:
fix expected tests outcomes

tag6071 committed:
More documentation to make code base descriptive

tag6071 committed:
Merge branch 'develop' into code-documentation

tag6071 committed:
Merge pull request #45 from ekundayo-ab/code-documentation

Code documentation

tag6071 committed:
refactor and fix errors in returning logic

tag6071 committed:
debugging travis build

tag6071 committed:
Merge pull request #46 from ekundayo-ab/code-documentation

Code documentation

tag6071 committed:
Debugging code reporting, test coverage and continous integration

tag6071 committed:
Badges implementation debugging

tag6071 committed:
Debugging badges implementation

tag6071 committed:
Debugging badges implementation

tag6071 committed:
Debugging badges implementation

tag6071 committed:
Debugging coveralls build

tag6071 committed:
Debugging coveralls badge implementation

tag6071 committed:
Debugging travis build

tag6071 committed:
Debugging travis badge build

tag6071 committed:
Debugging travis badge build

tag6071 committed:
Debugging travis and other test coverage reporting

tag6071 committed:
configure for travis build

tag6071 committed:
Debugging travis build

tag6071 committed:
Debug travis build

tag6071 committed:
Debug travis build

tag6071 committed:
Debug travis build more

tag6071 committed:
Debug travis

tag6071 committed:
Point badges update to develop branch

tag6071 committed:
Debug travis build the more

tag6071 committed:
removed coveralls token file

tag6071 committed:
Merge pull request #47 from ekundayo-ab/code-documentation

Debugging code reporting, test coverage and continous integration

tag6071 committed:
Update readme

tag6071 committed:
Merge pull request #48 from ekundayo-ab/code-documentation

Update readme

tag6071 committed:
Hotfix for codeclimate badge

tag6071 committed:
Merge pull request #49 from ekundayo-ab/code-documentation

Hotfix for codeclimate badge

tag6071 committed:
Debug codeclimate build

tag6071 committed:
Merge pull request #50 from ekundayo-ab/code-documentation

Debug codeclimate build

tag6071 committed:
Debugging codeclimate badge build

tag6071 committed:
Debug codeclimate build

tag6071 committed:
Merge pull request #51 from ekundayo-ab/code-documentation

Code documentation

tag6071 committed:
Add codeclimate coverage and issue badge

tag6071 committed:
Disable max len check by codeclimate because of materialize framework

tag6071 committed:
Merge pull request #52 from ekundayo-ab/code-documentation

Code documentation

tag6071 committed:
Debugging code climate build

tag6071 committed:
Merge pull request #53 from ekundayo-ab/code-documentation

Debugging code climate build

tag6071 committed:
Create .eslintignore

To ignore template folder containing materialize framework with

tag6071 committed:
Merge pull request #54 from ekundayo-ab/eslint-ignore-template-folder

Create .eslintignore

tag6071 committed:
Update .eslintignore

Debugging for travis build

tag6071 committed:
Update .eslintignore

Debugging for **codeclimate** build

tag6071 committed:
Merge pull request #55 from ekundayo-ab/eslint-ignore-template-folder

Eslint ignore template folder

tag6071 committed:
fix trailing white space issue

tag6071 committed:
surpress package import expressions errors by codeclimate

tag6071 committed:
Merge pull request #56 from ekundayo-ab/fix-codeclimate-issues

fix trailing white space issue

tag6071 committed:
remove trailing space as detected by codeclimate

tag6071 committed:
Merge pull request #57 from ekundayo-ab/fix-codeclimate-issues

remove trailing space as detected by codeclimate

tag6071 committed:
Update README.md

tag6071 committed:
Merge pull request #58 from ekundayo-ab/fix-readme-typos

Update README.md

tag6071 committed:
Setup swagger to document API routes

tag6071 committed:
Create swagger API documentation for various needed routes

tag6071 committed:
Add api-docs folder with all it's javascript to be ignored by ESLint

tag6071 committed:
Cleanups

tag6071 committed:
Merge pull request #59 from ekundayo-ab/swagger-api-documentation

Swagger api documentation

tag6071 committed:
Change swaggerDefinition to detect heroku port

tag6071 committed:
Merge pull request #60 from ekundayo-ab/swagger-api-documentation

Change swaggerDefinition to detect heroku port

tag6071 committed:
Point swagger-ui url to heroku url

tag6071 committed:
Debugging swagger documentation on heroku

tag6071 committed:
Debugging swagger api

tag6071 committed:
Merge pull request #61 from ekundayo-ab/swagger-api-documentation

Swagger api documentation

tag6071 committed:
Debugging heroku api still

tag6071 committed:
Debugging swagger api on heroku

tag6071 committed:
Debugging swagger api

tag6071 committed:
Still debugging swagger-api on heroku

tag6071 committed:
Debugging swagger api

tag6071 committed:
Swagger api debug on

tag6071 committed:
Swagger-ui api debugging heroku, refactored to latest version

tag6071 committed:
Swagger UI api debugging

tag6071 committed:
Debugging swagger-api on heroku

tag6071 committed:
Debugging heroku swagger-api

tag6071 committed:
Debugging heroku swagger-api documentation

tag6071 committed:
Merge pull request #62 from ekundayo-ab/swagger-api-documentation

Swagger api documentation

tag6071 committed:
Cleaning up to make API more descriptive

tag6071 committed:
Merge pull request #63 from ekundayo-ab/swagger-api-documentation

Cleaning up

tag6071 committed:
Cleaning up adding meta information to API documentation

tag6071 committed:
API hosting on heroku debugging

tag6071 committed:
Merge pull request #64 from ekundayo-ab/swagger-api-documentation

Swagger api documentation

tag6071 committed:
Add page title to API documentation page

tag6071 committed:
Merge pull request #65 from ekundayo-ab/swagger-api-documentation

Add page title to API documentation page

tag6071 committed:
Point API documentation page to live heroku domain

tag6071 committed:
Point API documentation page to live heroku domain

tag6071 committed:
Merge branch 'swagger-api-documentation' of github.com:ekundayo-ab/hello-books into swagger-api-documentation

tag6071 committed:
Temporal change of swagger-ui location to heroku live

tag6071 committed:
Merge pull request #66 from ekundayo-ab/swagger-api-documentation

Swagger api documentation

tag6071 committed:
Fix right host for api documentation

tag6071 committed:
Merge pull request #67 from ekundayo-ab/swagger-api-documentation

Fix right host for api documentation

tag6071 committed:
Debugging Swagger Docs on local

tag6071 committed:
More validation in borrowing and returning logic, fix API documentation not borrowing and returning bug

tag6071 committed:
Merge pull request #68 from ekundayo-ab/swagger-api-documentation

Swagger api documentation

tag6071 committed:
Debugging heroku deployment

tag6071 committed:
Debugging heroku deployment

tag6071 committed:
Still debugging heroku deployment

tag6071 committed:
Allow Cross Origin Requests to be served

tag6071 committed:
Merge pull request #69 from ekundayo-ab/swagger-api-documentation

Debugging heroku deployment

tag6071 committed:
Allow header for CORS as pertained to API documentation

tag6071 committed:
Merge pull request #70 from ekundayo-ab/swagger-api-documentation

Allow header for CORS

tag6071 committed:
Still debugging API documentation CORS

tag6071 committed:
Merge pull request #71 from ekundayo-ab/swagger-api-documentation

Still debugging API documentation CORS

tag6071 committed:
Add more API tests and wrote Middleware tests

tag6071 committed:
Adds more tests for unique email and username on user signup

tag6071 committed:
Temporal config changes for continuous integration

tag6071 committed:
Merge pull request #72 from ekundayo-ab/additional-api-tests

Additional api tests

tag6071 committed:
Refactor API & Middleware tests

tag6071 committed:
More tests for Book adding feature with neccessary refactoring

tag6071 committed:
Add more test cases for book borrowing and book modification API routes

tag6071 committed:
Add more test cases for Viewing all books, Deleting books and Returning of books

tag6071 committed:
Adds test cases for listing of books borrowed but not returned

tag6071 committed:
Temporal config file settings change for travis build

tag6071 committed:
Fix codeclimate issue by eslint

tag6071 committed:
Merge pull request #73 from ekundayo-ab/additional-api-tests

Additional api tests

tag6071 committed:
Abstracts Admin priviledge checking and Book adding and updating logic to helper functions

tag6071 committed:
More tests for helper functions used in validations and priviledge checking

tag6071 committed:
Merge pull request #74 from ekundayo-ab/additional-api-tests

Abstracts functional logics for some book operations

tag6071 committed:
Merge pull request #75 from ekundayo-ab/function-tests

More tests for helper functions

tag6071 committed:
Points coverage test path correctly after refactoring

tag6071 committed:
Merge pull request #76 from ekundayo-ab/function-tests

Points coverage test path correctly after refactoring

tag6071 committed:
Tests cases for User model

tag6071 committed:
Merge pull request #77 from ekundayo-ab/function-tests

Tests cases for User model

tag6071 committed:
Refactors storing important credentials in environment

tag6071 committed:
Travis debugging

tag6071 committed:
Debugging Travis

tag6071 committed:
Travis debugging on

tag6071 committed:
Debugging Travis

tag6071 committed:
Still debugging travis

tag6071 committed:
Debug travis

tag6071 committed:
Debug travis still

tag6071 committed:
Debugging travis

tag6071 committed:
Debug travis

tag6071 committed:
Debug travis

tag6071 committed:
Travis debugging on

tag6071 committed:
Merge pull request #79 from ekundayo-ab/function-tests

Function tests

tag6071 committed:
Refactors helpers for books and users into proper objects

tag6071 committed:
Makes isbn field unique

tag6071 committed:
Test cases for User, Book and Borrow models

tag6071 committed:
Merge pull request #80 from ekundayo-ab/model-tests

Model tests

tag6071 committed:
Refactor shelf page for responsiveness & less verbosity

tag6071 committed:
Refactors History & Shelf Page

tag6071 committed:
Refactor Admin page

tag6071 committed:
Refactor book detail page

tag6071 committed:
Refactor Profile page

tag6071 committed:
Links logo header, resize image to thumbnails on mobile

tag6071 committed:
Aligns user profile image with other layouts

tag6071 committed:
Change landing page background

tag6071 committed:
Merge pull request #81 from ekundayo-ab/template-refactoring

Template refactoring

tag6071 committed:
Implements custom styling in SASS

tag6071 committed:
Merge pull request #82 from ekundayo-ab/sass-custom-styling

Implements custom styling in SASS

tag6071 committed:
Ignores .DS_Store file

tag6071 committed:
Creates needed components for landing page

tag6071 committed:
Makes eslint ignores public folder in Client

tag6071 committed:
Merge pull request #83 from ekundayo-ab/components-with-react

Components with react

tag6071 committed:
Creates client routes for Signup, Login and Register

tag6071 committed:
Makes Register tab active by default visit to homepage

tag6071 committed:
Fixes CSS lint errors and formats html

tag6071 committed:
Fixes errors pointed out by hound

tag6071 committed:
Fixing errors pointed by hound

tag6071 committed:
Fixes issues pointed out by hound-ci

tag6071 committed:
Fixing to eliminate codeclimate issues

tag6071 committed:
resolving hound issues

tag6071 committed:
Fixing hound issues still

tag6071 committed:
fixing hound issues

tag6071 committed:
Merge pull request #84 from ekundayo-ab/components-with-react

Creates client routes for Signup, Login and Register

tag6071 committed:
Debugging hound issues

tag6071 committed:
Debugging hound-ci issues

tag6071 committed:
Fixes hound-ci issues

tag6071 committed:
More fixes for hound ci issues

tag6071 committed:
Debugging hound-ci issues

tag6071 committed:
Merge pull request #85 from ekundayo-ab/debug-custom-styling

Debugging hound issues

tag6071 committed:
Refactors present component to use pure webpack

tag6071 committed:
Defines proptypes properly in Landing Home component

tag6071 committed:
Merge pull request #86 from ekundayo-ab/setup-webpack-purely

Refactors present component to use pure webpack

tag6071 committed:
Creates signup form component, adds server side validation, adds client side validation

tag6071 committed:
able to redirect after signup

tag6071 committed:
Fixing conflicts and merging with develop

tag6071 committed:
Debugging conflicts fix

tag6071 committed:
Fixes missappropriation causing one test to fail

tag6071 committed:
Merge branch 'develop' into components-with-react

tag6071 committed:
Cleaning up after recovering from conflict

tag6071 committed:
Cleaning ups

osule commented on client/public/fonts/chunkfive/SIL Open Font License.txt:
```diff
@@ -0,0 +1,41 @@
+This Font Software is licensed under the SIL Open Font License, Version 1.1.
```
Can you get this font from a cdn?

osule commented on client/src/helpers/index.js:
```diff
@@ -0,0 +1,105 @@
+  static inputValidation(req) {
+    const errors = {};
+    for (let i = 0; i < 5; i += 1) {
+      const field = Object.values(req.body)[i];
```
You probably want to use a `let` here

tag6071 replied:
Ok thanks. *Eslint* with *airbnb* style guide which we were asked to use, actually detects it and recommends to change it to `const` @osule

osule commented on client/src/helpers/index.js:
```diff
@@ -0,0 +1,105 @@
+    const errors = {};
+    for (let i = 0; i < 5; i += 1) {
+      const field = Object.values(req.body)[i];
+      if (field === (undefined || null || '') || /^\s+$/.test(field)) {
```
What are you checking for here?

tag6071 replied:
I'm validating the input values when adding a book or modifying a book with respect to the database.

osule replied:
What are valid values for this field?

osule commented on client/src/helpers/index.js:
```diff
@@ -0,0 +1,105 @@
+    const errors = {};
+    for (let i = 0; i < 4; i += 1) {
+      const field = Object.values(req)[i];
+      if (field === (undefined || null || '') || /^\s+$/.test(field)) {
```
What is this checking for?

tag6071 replied:
This validates users details on registration

osule commented on server/api-docs/oauth2-redirect.html:
```diff
@@ -0,0 +1,53 @@
+        } else {
+            oauth2.callback({auth: oauth2.auth, token: qp, isValid: isValid, redirectUrl: redirectUrl});
+        }
+        window.close();
```
Which window is closed here?

tag6071 replied:
Actually this file was generated by swagger, I'll delete and see if it does not affect the swapper settings.

osule commented on server/config/config.js:
```diff
@@ -0,0 +1,21 @@
+    host: process.env.DEV_HOST,
+    dialect: 'postgres',
+  },
+  test: {
```
I don't think this should be here?

tag6071 replied:
Ok, where can I put it, as the real values are taken from an env file available locally and can be made available in travis, heroku environment

osule replied:
You may need to rethink environment variable names. 

Since you're using a similar database in all environments, it makes sense to me to have a single environment variable for the database connection. Each environment will set to a different value. e.g.

`DATABASE_URL`

osule commented on server/middlewares/auth.js:
```diff
@@ -1,31 +1,33 @@
+    }
     if (token) {
-      jwt.verify(token, 'hello-books', (err, decoded) => {
+      jwt.verify(token, process.env.JWTSECRET, (err, decoded) => {
```
Use underscores for spaces in variable names e.g. `JWT_SECRET`

osule commented on template/js/main.js:
```diff
@@ -2,4 +2,16 @@ $(document).ready(function(){
+function ConfirmDelete()
+{
+  var x = confirm("Are you sure you want to delete?");
+  if (x)
```
The variable can be better named.

tag6071 committed:
Creates more components as needed

tag6071 committed:
Adds uniqueness and data inputs validation

tag6071 committed:
Adds login global validations and input validations

tag6071 committed:
Adds password matching validation on both server and client side

tag6071 committed:
Adds password matching validation on both server and client side

olawalejarvis commented on .babelrc:
```diff
@@ -1,3 +1,3 @@
 {
-    "presets":["es2015", "stage-2"]
+    "presets":["es2015"]
 }
```
The danger symbol signifies that a new line is required

olawalejarvis commented on .codeclimate.yml:
```diff
@@ -1,7 +1,16 @@
+  paths:
+  - "**.js"
+exclude_paths:
+- server/tests/
```
New line

tag6071 replied:
Duly noted

olawalejarvis commented on .travis.yml:
```diff
@@ -1,13 +1,36 @@
+services:
+  - postgresql
 before_script:
   - npm install -g codeclimate-test-reporter
```
Why two separate "before_script" (line 6 and 24)?

tag6071 replied:
No reason, I'm cleaning it up.

olawalejarvis commented on .travis.yml:
```diff
@@ -1,13 +1,36 @@
\ No newline at end of file
+  - npm run coverage
+  - codeclimate-test-reporter < coverage/lcov.info
+  - coveralls < coverage/lcov.info
```
New line is required here

tag6071 replied:
Yes. Would implement

olawalejarvis commented on client/public/js/main.js:
```diff
@@ -0,0 +1,19 @@
+$(document).ready(function(){
```
Why are not using reactjs for this?

tag6071 replied:
Didn't originally think to use it, I would see how to implement with reactjs

olawalejarvis commented on client/src/components/flash/FlashMessage.js:
```diff
@@ -0,0 +1,36 @@
+  }
+
+  render() {
+    const { /* id, */ type, text } = this.props.message;
```
You might want to remove `id` if you're not using it

tag6071 committed:
Refactors and makes all tests pass after

tag6071 committed:
Merge branch 'components-with-react' of github.com:ekundayo-ab/hello-books into components-with-react

tag6071 committed:
Redirects to shelf on login or register

tag6071 committed:
Creates a category model with adding of category feature

tag6071 committed:
Simplifies Admin page into components

tag6071 committed:
Starts implementation for adding book

tag6071 committed:
Switching to setting up without create-react-app

tag6071 committed:
Switching to setting up without create-react-app

tag6071 committed:
Cleanups

tag6071 committed:
Cleanups and makes swagger loads ui from `/api/v1` appended to root url

tag6071 committed:
Finally refactors api documentation to `/api/docs`

tag6071 committed:
Implementing Feedbacks

- Splits up **.babelrc** arrays element each on newline
- Refactors two **before_scripts** into one in **travis.yml**
- Fetches font **Chunkfive** font from CDN and put in **client/index.html** head
- Renames variable to confirm delete action to be more descriptive
- Refactors jwt token secret variable to use underscore for spaces **JWT_SECRET**

tag6071 committed:
Starts integrating react properly with webpack

tag6071 committed:
Configures webpack independently now, not using create-react-app scaffold

tag6071 committed:
Refactors client side signin & signup
- Validations for data entered in signin/signup form
- Flash messages for errors
- Flash messages for success after registering or login in

osule commented on client/src/actions/authActions.js:
```diff
@@ -12,18 +8,28 @@ function setCurrentUser(user) {
+*/
+
+function login(data) {
+  return () => axios.post('/api/v1/users/signin', data);
```
Why is this returning a function?

tag6071 replied:
to abstract the database request (action)...with the promise available to be worked on in the component that needs it.

osule commented on client/src/actions/authActions.js:
```diff
@@ -12,18 +8,28 @@ function setCurrentUser(user) {
+}
+
+function userSignUpRequest(userData) {
+  return () => axios.post('/api/v1/users/signup', userData);
```
Same comment as above here

osule commented on client/src/components/Landing/SigninForm.js:
```diff
@@ -38,9 +38,20 @@ class SignIn extends Component {
+            text: 'Oops! Something happened try again or contact us',
+          });
+        },
         (err) => {
```
You should probably have this in a `catch`

osule commented on client/src/components/library/admin/BookForm.js:
```diff
@@ -2,7 +2,7 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { saveBook } from './../../../actions/createActions';
+import { saveBook } from './../../../actions/bookActions';
```
Move `actions` to component directory.

`import {saveBook} from `/app/components/Book/actions``

tag6071 replied:
ok. so actions folder should be a subfolder in components directory right?

osule replied:
That's right.

osule commented on client/src/components/library/admin/BookForm.js:
```diff
@@ -21,7 +21,7 @@ class BookForm extends Component {
+  onChange(e) {
     if (!this.state.errors[e.target.name]) {
       const errors = Object.assign({}, this.state.errors);
       delete errors[e.target.name];
```
Why is this happening?

tag6071 replied:
to ensure an error is not populated again for a field if there are no errors after correction

tag6071 committed:
feature(components-with-react):

- Add Signing Up functionality
- Add Signing In functionality
- Set current user on login
- Unset current user on logout
